File: 108.将有序数组转换为二叉搜索树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return sortedArrayToBST(nums, 0, nums.size() - 1);
    }
    
    TreeNode* sortedArrayToBST(vector<int>& nums, int left, int right){
        if(left > right)
            return NULL;
        int mid = left + (right - left) / 2;
        TreeNode* temp = new TreeNode(nums[mid]);
        temp->left = sortedArrayToBST(nums, left, mid - 1);
        temp->right = sortedArrayToBST(nums, mid + 1, right);
        return temp;
    }
};

File: 二叉树/108.将有序数组转换为二叉搜索树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
 var sortedArrayToBST = function(nums) {
    var n = nums.length;
    
    const createTree = (nums, l, r) => {
        if (l > r)
            return null;
        let mid = Math.floor((l + r + 1) / 2);
        let node = new TreeNode(nums[mid]);
        node.left = createTree(nums, l, mid - 1);
        node.right = createTree(nums, mid + 1, r);
        return node;
    }

    return createTree(nums, 0, n - 1);
};